packages/icon/package.py

```diff
@@ -1,294 +1,710 @@
 
-    def setup_build_environment(self, env):
-        self.config_dir = self.spec.variants['config_dir'].value
-        _config_file_name = self.spec.variants[
-            'host'].value + '.' + self.spec.variants['icon_target'].value
-        if self.compiler.name == 'cce':
-            _config_file_name += '.cray'
-        elif self.compiler.name == 'nvhpc' or self.compiler.name == 'pgi':
-            _config_file_name += '.nvidia'
-        else:
-            _config_file_name += '.' + self.compiler.name
-
-        self._config_file_name = _config_file_name
-
-        if '+dace' in self.spec:
-            env.set('ICON_FCFLAGS', '-O2')
-            env.set('ICON_DACE_FCFLAGS', '-O1')
-
-        if '~skip-config' in self.spec:
-            env.set('XML2_ROOT', self.spec['libxml2'].prefix)
-            if self.spec.variants['serialize_mode'].value != 'none':
-                env.set('SERIALBOX2_ROOT', self.spec['serialbox'].prefix)
-            if '+claw' in self.spec:
-                env.set('CLAW', self.spec['claw'].prefix + '/bin/clawfc')
-            if '+eccodes' in self.spec:
-                env.set('ECCODES_ROOT', self.spec['eccodes'].prefix)
-        if self.run_tests:
-            # setting BB_SYSTEM sets d56 as account in file create_target_header
-            env.set('BB_SYSTEM', 'use_d56_account')
+    serialization_values = ('read', 'perturb', 'create')
+    variant('serialization',
+            default='none',
+            values=('none', ) + serialization_values,
+            description='Enable the Serialbox2 serialization')
 
-    def configure_args(self):
+    # Optimization Features:
+    variant('mixed-precision',
+            default=False,
+            description='Enable mixed precision dycore')
 
-        args = []
+    depends_on('libxml2', when='+coupling')
+    depends_on('libxml2', when='+art')
 
-        # Icon-hammoz:
-        if '+ham' in self.spec:
-            args.append('--enable-atm-phy-echam-submodels')
-            args.append('--enable-hammoz')
+    depends_on('rttov+hdf5', when='+rttov')
+    depends_on('rttov~openmp', when='~openmp+rttov')
 
-        # Serialization
-        if self.spec.variants['serialize_mode'].value != 'none':
-            args.append('--enable-serialization=' +
-                        self.spec.variants['serialize_mode'].value)
+    for x in serialization_values:
+        depends_on('serialbox+fortran~shared',
+                   when='serialization={0}'.format(x))
 
-        # Art
-        if '+art' in self.spec:
-            args.append('--enable-art')
+    depends_on('libcdi-pio+fortran+netcdf', when='+cdi-pio')
+    depends_on('libcdi-pio grib2=eccodes', when='+cdi-pio+grib2')
+    depends_on('libcdi-pio+mpi', when='+cdi-pio+mpi')
 
-        # Claw
-        if '+claw' in self.spec:
-            args.append('--enable-claw')
+    # needed to mimick BB in test.py
+    # only available in icon-exclaim so far
+    depends_on('cdo', when='@exclaim-master,exclaim-test')
```

Comment:
`depends_on('cdo', when='@exclaim-master,exclaim-test')` means cdo is a build and link dependency, but I think it's a run dependency. Right?